nuxt.config.ts

```diff
@@ -1,12 +1,47 @@
-import { defineNuxtConfig } from 'nuxt'
+import { defineNuxtConfig } from "nuxt";
+import AutoImport from "unplugin-auto-import/vite";
+import Components from "unplugin-vue-components/vite";
+import { ElementPlusResolver } from "unplugin-vue-components/resolvers";
 
 // https://v3.nuxtjs.org/api/configuration/nuxt.config
 export default defineNuxtConfig({
-    css: ["element-plus/dist/index.css"],
-    build: {
-        transpile: [
-            '@apollo/client',
-            'ts-invariant/process',
-        ],
+  publicRuntimeConfig: {
+    API_ENDPOINT: process.env.NUXT_ENV_API_ENDPOINT
+  },
+  head: {
+    title: "nuxt3-GraphQL-pet",
+    htmlAttrs: {
+      lang: "en"
     },
-})
+    meta: [
+      { charset: "utf-8" },
+      { name: "viewport", content: "width=device-width, initial-scale=1" },
+      { hid: "description", name: "description", content: "" },
+      { name: "format-detection", content: "telephone=no" }
+    ]
+  },
+  vite: {
+    plugins: [
+      // ...
+      AutoImport({
+        resolvers: [ElementPlusResolver()]
+      }),
+      Components({
+        resolvers: [ElementPlusResolver()]
+      })
+    ],
+    ssr: {
+      noExternal: ["element-plus"]
+    }
+  },
+  css: [
+    "element-plus/dist/index.css",
```

Comment:
you don't need to import that element-plus/dist/index.css, you have import on demand here
```
vite: {
    plugins: [
      // ...
      AutoImport({
        resolvers: [ElementPlusResolver()]
      }),
      Components({
        resolvers: [ElementPlusResolver()]
      })
    ],
    ssr: {
      noExternal: ["element-plus"]
    }
  },
```